hedera-node/src/main/java/com/hedera/services/context/SideEffectsTracker.java

```diff
@@ -342,21 +347,32 @@ public void trackTokenUnitsChange(final TokenID token, final AccountID account,
 	 */
 	public void trackNftOwnerChange(final NftId nftId, final AccountID from, AccountID to) {
 		final var token = nftId.tokenId();
-		tokensTouched[numTouches++] = token;
+		tokensTouched[numTokenChangesSoFar++] = token;
 		var xfers = nftOwnerChanges.computeIfAbsent(token, ignore -> TokenTransferList.newBuilder());
 		xfers.addNftTransfers(nftTransferBuilderWith(from, to, nftId.serialNo()));
 	}
 
 	/**
 	 * Returns the list of net ℏ balance changes including all incremental side effects tracked since the
-	 * last call to {@link SideEffectsTracker#reset()}. The returned list is sorted in ascending order by
-	 * the {@link com.hedera.services.ledger.HederaLedger#ACCOUNT_ID_COMPARATOR}.
+	 * last call to {@link SideEffectsTracker#reset()}. The returned adjustments are sorted in ascending order of
+	 * account numbers
 	 *
 	 * @return the ordered net balance changes
 	 */
-	public TransferList getNetTrackedHbarChanges() {
-		purgeZeroAdjustments(netHbarChanges);
-		return netHbarChanges.build();
+	public CurrencyAdjustments getNetTrackedHbarChanges() {
+		numHbarChangesSoFar = purgeZeroChanges(changedAccounts, balanceChanges, numHbarChangesSoFar);
+
+		// copy the range of elements that are modified from balance changes and account numbers
+		final long[] changedBalances;
+		final long[] changedAccountNums;
+		if (numHbarChangesSoFar < balanceChanges.length) {
```

Comment:
We don't want this check---we _always_ want to pass copied arrays into the `CurrencyAdjustments.fromChanges()` factory.

(If the the `ExpirableTxnRecord` with this `CurrencyAdjustments` shared the same array as the `SideEffectsTracker` singleton, then the next transaction being handled might change that record's transferlist while it was being serialized for hashing---or being converted to gRPC for the record stream!)

Reply:
Sorry misunderstood the previous comment. 
Removed the check and always copied the range of changes to new arrays before passing to `CurrencyAdjustments`.